### DISCUSSION TITLE
Can't  execute hive sql include chinese

### DISCUSSION THREAD

**Discussion by gzqcason** [cloudera/hue#2820] [Questions] [answered]
Dear all
as you can see the statement can't includ Chinese，
![image](https://user-images.githubusercontent.com/79817470/164966599-013df869-abc3-46c0-bef6-3f2265c03b15.png)

I checked the metadata in MySQL, the Chinese parsed into unicode and kept at table ‘desktop_docyment2’
![cd5b26600a8c2fca9d9d2550bbfa7c8](https://user-images.githubusercontent.com/79817470/164966763-1547e3e5-526c-451e-95e3-d99fce105f47.png)

and then I use unicode directly, Hue execute  good. I don't know how to solve this problem, need help, thanks a lot.
![image](https://user-images.githubusercontent.com/79817470/164966885-e4cd740f-8e17-4dba-936a-db97360fb1a2.png)

btw, I use open source hue-4.10 version and hive-2.7. **I changed encoding in hue's metatable(make chinese can show on web), and changed python default encoding into utf-8**. However I guess it's only something woring in parse-execute stage.

**Comment by Akhilsnaik** [comment]
do you know what is there in the logs ?

probably hitting : https://github.com/cloudera/hue/pull/1903 ?

**Reply by gzqcason** [reply]
As i rerstart Hue like ‘nohup supervisor > test.log 2>&1 &’ , got same error info in 'hue/logs/error.log' while execute sql :
thrift_util and decorators: QueryError: Error while compiling statement: FAILED: ParseException line 1:9 cannot recognize input near '<EOF>' '<EOF>' '<EOF>' in select clause

run the test shell got info below:
![image](https://user-images.githubusercontent.com/79817470/165283871-7712417d-1af3-4b06-941c-dc7d0ed4dce4.png)


met similar problem befor I changed python2.7's default encoding (from ascii to utf8) ：
‘Processing exception: 'ascii' codec can't encode characters in position 68-69: ordinal not in range(128)’

**Reply by 2416210017** [reply]
Have you solved it? I also report the same error here. Hue version 4.10

![image](https://user-images.githubusercontent.com/52597892/222094592-e715ad9e-28c0-40fd-a95d-967bb53dd242.png)

![image](https://user-images.githubusercontent.com/52597892/222094549-f2f43d9a-0942-479c-aa5a-7c383aced16e.png)

**Reply by 2416210017** [reply]
vim /data/hue-4.10/hue/apps/beeswax/gen-py/TCLIService/ttypes.py

oprot.writeString(self.statement) 编辑为 oprot.writeString(self.statement.encode('utf-8'))

Here is the answer:
https://github.com/cloudera/hue/issues/2606

![image](https://user-images.githubusercontent.com/52597892/222106819-dd89898c-e202-41e6-b151-c408dc8f8dc4.png)

**Comment by Harshg999** [comment, marked as answer]
> vim /data/hue-4.10/hue/apps/beeswax/gen-py/TCLIService/ttypes.py
> 
> oprot.writeString(self.statement) 编辑为 oprot.writeString(self.statement.encode('utf-8'))
> 
> Here is the answer: #2606
> 
> ![image](https://user-images.githubusercontent.com/52597892/222106819-dd89898c-e202-41e6-b151-c408dc8f8dc4.png)


The fix is now also part of the latest 4.11.0 release.
For earlier versions, fix can be applied as described in: https://github.com/cloudera/hue/issues/2606

Thanks @2416210017 for pointing out!
